conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceInterfaceGenerator.java

```diff
@@ -96,17 +99,69 @@ private MethodSpec generateServiceInterfaceMethod(
 
         endpointDef.getDeprecated().ifPresent(deprecatedDocsValue -> methodBuilder.addAnnotation(Deprecated.class));
 
-        methodBuilder.addJavadoc("$L", ServiceGenerators.getJavaDocWithRequestLine(endpointDef));
+        addJavaDoc(methodBuilder, endpointDef, options.packagePrefix());
 
         if (UndertowTypeFunctions.isAsync(endpointDef, options)) {
             methodBuilder.returns(UndertowTypeFunctions.getAsyncReturnType(endpointDef, returnTypeMapper, options));
         } else {
             endpointDef.getReturns().ifPresent(type -> methodBuilder.returns(returnTypeMapper.getClassName(type)));
         }
 
+        methodBuilder.addExceptions(endpointDef.getErrors().stream()
+                .map(endpointError -> {
+                    ErrorTypeName errorTypeName = endpointError.getError();
+                    return ClassName.get(
+                            Packages.getPrefixedPackage(errorTypeName.getPackage(), options.packagePrefix()),
+                            "Server" + errorTypeName.getNamespace() + "Errors",
```

Comment:
It might be helpful to take all the occurrences of `"Server" + namespace + "Errors"` and consolidate them into a utility function.

I think we're at the point in the process where I question my instincts on naming (which I'm the first to admit I'm truly terrible at!), and wonder if we should opt for `namespace + "ServerErrors"` so that namespace provides tab completion. What do you think? (tagging @bjlaub and @bmarcaur to opine on this as well).

The other piece to consider is collision between the existing and server-side error utilities: If you have a namespace called `ServerFoo` and another called `Foo`, you'd end up with two classes called `ServerFoo` potentially with the same fully qualified package name. The fix is pretty simple, though, a java wire break can be avoided by changing the `package` of one of the two, but it's a bit of friction that would be nice to avoid if possible.

Reply:
We sync'd earlier and came to the conclusion that we should use `Server<namespace>Errors`. Though now, I I like the discoverability that `<namespace>ServerErrors` provides. The only advantage of prefixing the current errors class (i.e. using `Server<namespace>Errors`) was to make migrations slightly less painful.

I'll update.